app/services/katello/candlepin/consumer.rb

```diff
@@ -99,11 +99,13 @@ def pool_ids
       end
 
       def virtual_guests
+        return [] if self.uuid.nil?
```

Comment:
There were candlepin.log errors triggered when making these calls during the host create via api since there is no uuid yet.